### ISSUE TITLE
Keep terminal content after reconnecting (SSH)

### ISSUE THREAD

**Issue by jehon** [microsoft/vscode#95183]
Using VSCode insider, we have the great "remote SSH", working on a remote server.

When the connection is lost, VSCode propose to reconnect to the server. Great.
But when it does, we loose all the terminal windows, and more important, all the content of those windows.

I would appreciate if the "content" of the terminal windows would remain, even if the tab is not connected anymore... Something like a "Dead pane"... allowing us to see what did happen in the terminal while reloading the window.

Have a nice day

**Comment by Tyriar** [comment]
Related: https://github.com/microsoft/vscode/issues/20013

**Comment by jehon** [comment]
Yes indeed, if 20013 is implemented, this could be solved too...

This one is a bit more simple, in a way that I would like only the content to be kept (ex: to see the result of a long run), and not the connection itself...

**Comment by Tyriar** [comment]
@jehon keeping the content is actually one of the harder parts, luckily it looks like we've started prioritizing #20013 as a long term goal and it's on this plan with this item:

> xterm.js: Prepare client/server separation for terminals https://github.com/xtermjs/xterm.js/issues/2749

https://github.com/microsoft/vscode/issues/95168

**Comment by JavaCS3** [comment]
xterm-addon-serialize can solve this problem

**Comment by jehon** [comment]
Thinking about this, one acceptable solution for me would be to automatically start a "screen" session when starting the terminal, and to reconnect to it when restarting the vs-code...

**Comment by Tyriar** [comment]
I have plans for this such that it should work on Windows, with any shell and not needing outside dependencies. Part of this was even on the iteration plan last month pointing at https://github.com/xtermjs/xterm.js/issues/2749
